### ISSUE TITLE
Decipher - CUDA tests failure

### ISSUE THREAD

**Issue by ori-kron-wis** [scverse/scvi-tools#3109]
Hi @justjhong ,

We merged https://github.com/scverse/scvi-tools/pull/3091, and it didn't run CUDA tests.
On other branches that ran the merged decipher tests on CUDA, we got 2 failures:

![Image](https://github.com/user-attachments/assets/985f7b3f-2387-4096-be00-c9ea47baf285)

![Image](https://github.com/user-attachments/assets/acbcbddd-f9af-43ae-884b-dacf30c5edac)

See for example: https://github.com/scverse/scvi-tools/actions/runs/12458060589/job/34773264674?pr=3100

Basically its about GPU-CPU integration issues. 

You can go ahead and run CUDA tests on the PR fix: https://github.com/scverse/scvi-tools/pull/3110
